send.go

```diff
@@ -64,12 +64,36 @@ func (d *Data) EncodeABI(v []byte) []byte {
 
 func (d *Data) internal() {}
 
+// EncodeABI implements EtherType.EncodeABI
+func (b *Bytes) EncodeABI(v []byte) []byte {
+	return append(v, padright(*b)...)
+}
+
+// Len implements EtherSlice.Len
+func (b *Bytes) Len() int  { return len(*b) }
+func (b *Bytes) internal() {}
+
 func padint(i int, v []byte) []byte {
 	var w [32]byte
 	binary.BigEndian.PutUint64(w[32-8:], uint64(i))
 	return append(v, w[:]...)
 }
 
+func padright(b []byte) []byte {
+	mod := len(b) % 32
+	var padding int
+	if mod > 0 {
+		padding = 32 - mod
+	} else {
+		padding = 0
+	}
+
+	result := make([]byte, len(b)+padding)
```

Comment:
You can replace `len(b)+padding` with `(len(b)+31)&-32)` and remove the padding calculation

Reply:
Done.